packages/twenty-server/src/engine/core-modules/actor/services/created-by-from-auth-context.service.ts

```diff
@@ -1,23 +1,89 @@
 import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
 
 import { isDefined } from 'twenty-shared/utils';
+import { Repository } from 'typeorm';
 
 import { buildCreatedByFromApiKey } from 'src/engine/core-modules/actor/utils/build-created-by-from-api-key.util';
 import { buildCreatedByFromFullNameMetadata } from 'src/engine/core-modules/actor/utils/build-created-by-from-full-name-metadata.util';
 import { AuthContext } from 'src/engine/core-modules/auth/types/auth-context.type';
 import { ActorMetadata } from 'src/engine/metadata-modules/field-metadata/composite-types/actor.composite-type';
 import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
 import { WorkspaceMemberWorkspaceEntity } from 'src/modules/workspace-member/standard-objects/workspace-member.workspace-entity';
+import { FieldMetadataEntity } from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type RecordToCreateData = Record<string, any>;
 
 @Injectable()
 export class CreatedByFromAuthContextService {
   private readonly logger = new Logger(CreatedByFromAuthContextService.name);
 
   constructor(
+    @InjectRepository(FieldMetadataEntity, 'metadata')
+    private readonly fieldMetadataRepository: Repository<FieldMetadataEntity>,
     private readonly twentyORMGlobalManager: TwentyORMGlobalManager,
   ) {}
 
-  public async buildCreatedBy(
+  injectCreatedBy(
+    records: RecordToCreateData[],
+    objectName: string,
+    authContext: AuthContext,
+  ): Promise<RecordToCreateData[]>;
+
+  injectCreatedBy(
+    record: RecordToCreateData,
+    objectName: string,
+    authContext: AuthContext,
+  ): Promise<RecordToCreateData>;
+
+  async injectCreatedBy(
+    recordOrRecords: RecordToCreateData | RecordToCreateData[],
```

Comment:
proposal: would be nicer to always pass records and in case of create-one, to pass [record]